test/unit/identifiers.spec.ts

```diff
@@ -788,3 +788,39 @@ test("lua built-in as in constructor assignment", () => {
         export const result = new A("42").error;
     `.expectToMatchJsResult();
 });
+
+test("local variable declaration referencing self indirectly", () => {
```

Comment:
It doesn't seem to fit the `identifiers.spec.ts` test file, which generally tests identifier printing itself, not how it's used. I think it would make sense to test it along with self-calling function, or general assignment tests